CMakeLists.txt

```diff
@@ -9,32 +9,24 @@ if(NOT ENABLE_WEBSOCKET OR NOT ENABLE_UI)
 endif()
 
 # Submodule deps check
-if(NOT
-   (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/deps/json/CMakeLists.txt
-    AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/deps/websocketpp/CMakeLists.txt
-    AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/deps/qr/cpp/QrCode.hpp
-    AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/deps/asio/asio/include/asio.hpp))
+if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/deps/qr/cpp/QrCode.hpp)
   obs_status(FATAL_ERROR "obs-websocket submodule deps not available.")
 endif()
 
 # Plugin tests flag
 option(PLUGIN_TESTS "Enable plugin runtime tests" OFF)
 
-# Qt build stuff
-set(CMAKE_PREFIX_PATH "${QTDIR}")
-set(CMAKE_INCLUDE_CURRENT_DIR ON)
-set(CMAKE_AUTOMOC ON)
-set(CMAKE_AUTOUIC ON)
-set(CMAKE_AUTORCC ON) # For resources.qrc
-
 # Find Qt
 find_qt(COMPONENTS Core Widgets Svg Network)
 
-# Find nlohmann
-set(JSON_BuildTests
-    OFF
-    CACHE INTERNAL "")
-add_subdirectory(deps/json)
+# Find nlohmann JSON
+find_package(nlohmann_json 3 REQUIRED)
+
+# Find WebSocket++
+find_package(Websocketpp 0.8 REQUIRED)
+
+# Find Asio
+find_package(Asio 1.12.1 REQUIRED)
```

Comment:
Should this use the `EXACT` option to ensure that we do not accidentally pick up a different version?

Reply:
All Linux-related `find_package` should operate with the assumption that we can't enforce a exact version but only a minimum.

We control Windows, macOS and Flatpak deps version but not Ubuntu's (and other distros').

Reply:
As far as I know, the version limitations are mainly for MacOS and specifically for ASIO. I'm not aware of any intentional breaking changes in any recent history for any of these dependencies.

Reply:
I though we were talking about minimums on the three deps.

I placed them mostly to avoid building OBS with too old version of those libs mostly on old Linux distros.

Reply:
I can remove them, if you consider this is really unneeded.

Reply:
I think it's fair to keep them as is. I was mainly responding to whether or not we should use the `EXACT` keyword.

Reply:
It's fine. My question has been answered. We control the version on the system where it's problematic, and using `EXACT` would probably be too burdensome for non-Flatpack Linux builds.